From: Jason Dillaman <dillaman@redhat.com>
Date: Thu, 11 Aug 2016 21:11:07 -0400
Subject: test: fixed unused function warnings in unittest_rbd_mirror

Signed-off-by: Jason Dillaman <dillaman@redhat.com>
(cherry picked from commit 3545d9ed19be8d3956f0db901ea9d3bb8b10d13d)

Resolves: rhbz#1349955
---
 src/test/rbd_mirror/test_mock_ImageReplayer.cc      | 1 -
 src/test/rbd_mirror/test_mock_ImageSyncThrottler.cc | 1 -
 src/tools/rbd_mirror/ImageReplayer.cc               | 6 ------
 src/tools/rbd_mirror/ImageReplayer.h                | 2 --
 4 files changed, 10 deletions(-)

diff --git a/src/test/rbd_mirror/test_mock_ImageReplayer.cc b/src/test/rbd_mirror/test_mock_ImageReplayer.cc
index 5f651af..2477e03 100644
--- a/src/test/rbd_mirror/test_mock_ImageReplayer.cc
+++ b/src/test/rbd_mirror/test_mock_ImageReplayer.cc
@@ -168,7 +168,6 @@ ReplayStatusFormatter<librbd::MockTestImageCtx>* ReplayStatusFormatter<librbd::M
 
 // template definitions
 #include "tools/rbd_mirror/ImageReplayer.cc"
-template class rbd::mirror::ImageReplayer<librbd::MockTestImageCtx>;
 
 namespace rbd {
 namespace mirror {
diff --git a/src/test/rbd_mirror/test_mock_ImageSyncThrottler.cc b/src/test/rbd_mirror/test_mock_ImageSyncThrottler.cc
index e624ed9..772a49e 100644
--- a/src/test/rbd_mirror/test_mock_ImageSyncThrottler.cc
+++ b/src/test/rbd_mirror/test_mock_ImageSyncThrottler.cc
@@ -101,7 +101,6 @@ std::vector<MockImageSync *> MockImageSync::instances;
 
 // template definitions
 #include "tools/rbd_mirror/ImageSyncThrottler.cc"
-template class rbd::mirror::ImageSyncThrottler<librbd::MockTestImageCtx>;
 
 namespace rbd {
 namespace mirror {
diff --git a/src/tools/rbd_mirror/ImageReplayer.cc b/src/tools/rbd_mirror/ImageReplayer.cc
index 5ae89d8..565dced 100644
--- a/src/tools/rbd_mirror/ImageReplayer.cc
+++ b/src/tools/rbd_mirror/ImageReplayer.cc
@@ -705,12 +705,6 @@ void ImageReplayer<I>::on_stop_journal_replay()
 }
 
 template <typename I>
-void ImageReplayer<I>::close_local_image(Context *on_finish)
-{
-  m_local_image_ctx->state->close(on_finish);
-}
-
-template <typename I>
 void ImageReplayer<I>::handle_replay_ready()
 {
   dout(20) << "enter" << dendl;
diff --git a/src/tools/rbd_mirror/ImageReplayer.h b/src/tools/rbd_mirror/ImageReplayer.h
index 1b1a742..c0c60cf 100644
--- a/src/tools/rbd_mirror/ImageReplayer.h
+++ b/src/tools/rbd_mirror/ImageReplayer.h
@@ -199,8 +199,6 @@ protected:
 
   bool on_replay_interrupted();
 
-  void close_local_image(Context *on_finish); // for tests
-
 private:
   typedef typename librbd::journal::TypeTraits<ImageCtxT>::Journaler Journaler;
   typedef boost::optional<State> OptionalState;
